src/JestProcessManagement/JestProcess.ts

```diff
@@ -79,12 +80,14 @@ export class JestProcess {
   }
 
   public stop(): Promise<void> {
-    this.stopRequested = true
-    this.keepAliveCounter = 1
-    this.jestSupportEvents.clear()
     return new Promise(resolve => {
+      this.stopRequested = true
+      this.keepAliveCounter = 1
       this.resolve = resolve
+      this.jestSupportEvents.clear()
       this.runner.closeProcess()
+
+      setTimeout(resolve, JestProcess.stopHangTimeout)
```

Comment:
this looks clean, however, for the sake of future maintenance,  I am thinking we should
1. add some comment to brief why we are adding this timeout watch here
1. If we are serious about removing this in the future, we should have a way to tell if the timeout is "redundant", i.e. the promise has already been resolved.  Right now it doesn't seem to be able to tell us that. We can certainly refactor it to reset resolve, emit logging etc, but it will no doubt adding more pain for the already confusing state management here... hmmm... Let's at least add a TODO comment to outline what should be done, or create an issue to track all the issues we discussed here but deferred?

BTW, I assume you have build vscode-jest with this PR and tested in your dev environment, right? I haven't got a chance to do so, if you have done that then we can merge once the comment is added.

Again, thanks for your contribution 👍 , sorry for the delay but we are almost there...

Reply:
1. Do you have advice for how I should word the comment?
2. From the what I've found, [calling resolve additional times should be a no-op](https://stackoverflow.com/questions/20328073/is-it-safe-to-resolve-a-promise-multiple-times) ("If alreadyResolved.[[value]] is true, return undefined."). But if we wanted to avoid calling it anyway, we could perhaps set `this.resolve = null` after the normal `resolve()` and then wrap the timeout's resolve in a `if (this.resolve === resolve)` wrapper?

I have done some brief tests with running it in my dev environment, but I haven't done a serious amount of dogfooding with it yet.

Reply:
> Do you have advice for how I should word the comment?

put yourself in the future developers' shoes, when they see the promise is resolved in the callback functions, and then find a timeout watch here, wouldn't they be confused? Remeber the question we were asking when we saw the original timeout in JestExt?... I suggest adding  the "why", and a TODO to outline your ultimate vision... oh and feel free to link back to the relevant discussions in this PR, if needed.

> But if we wanted to avoid calling it anyway, we could perhaps set this.resolve = null after the normal resolve() and then wrap the timeout's resolve in a if (this.resolve === resolve) wrapper?

something like that, but giving there are already too many questionable ivar/state that we want to clean up, I really hate to see us piling up more... that's why I am open to tackle these state-management related issues in a separate refactoring PR, so long as we put some comments or file an issue to follow up...

Reply:
I've added some comments about the timeout, and the `this.resolve === resolve` sanity check. Hopefully, a future refactor can remove the need for the `this.resolve` ivar altogether, and make it more promise-based.